internal/terraform/datadir/terraform_sources.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package datadir
+
+import (
+	"fmt"
+	"io/fs"
+	"path/filepath"
+
+	"github.com/hashicorp/go-slug/sourcebundle"
+)
+
+var terraformSourcesDirElements = []string{
+	DataDirName, "modules",
+}
+var terraformSourcesPathElements = []string{
+	DataDirName, "modules", "terraform-sources.json",
+}
+
+func TerraformSourcesDirPath(fs fs.StatFS, modulePath string) (string, bool) {
+	terraformSourcesPath := filepath.Join(
+		append([]string{modulePath},
+			terraformSourcesPathElements...)...)
+	terraformSourcesDirPath := filepath.Join(
+		append([]string{modulePath},
+			terraformSourcesDirElements...)...)
+
+	fi, err := fs.Stat(terraformSourcesPath)
+	if err == nil && fi.Mode().IsRegular() {
+		return terraformSourcesDirPath, true // TODO: this is a bit weird and misleading, maybe we should just use the bundle thing reading the dir and catch the proper error or sth like that
```

Comment:
Unsure whether we need this method at all since the go-slug parser for this file also handles it not existing

Reply:
Good point. I think you can remove it and make `ParseTerraformSources` a bit leaner.